sklearn/grid_search.py

```diff
@@ -111,6 +113,44 @@ def __len__(self):
         return sum(product(len(v) for v in p.values()) if p else 1
                    for p in self.param_grid)
 
+    def __getitem__(self, ind):
+        """Get the parameters that would be ``ind``th in iteration
+
+        Parameters
+        ----------
+        ind : int
+            The iteration index
+
+        Returns
+        -------
+        params : dict of string to any
+            Equal to list(self)[ind]
+        """
+        for sub_grid in self.param_grid:
+            # XXX: could memoize information used here
+            if not sub_grid:
+                if ind == 0:
+                    return {}
+                else:
+                    ind -= 1
+                    continue
+            keys, values_lists = zip(*sorted(sub_grid.items()))
+            sizes = [len(v_list) for v_list in values_lists]
+            # For sizes = [5, 3, 2], modulo = [30,  6,  2,  1]
+            # The grid has size 30;
+            # the first param changes every 6, the second every 2, etc.
+            modulo = np.cumprod(np.hstack([1, sizes[::-1]]))[::-1]
+            if ind >= modulo[0]:
+                # Try the next grid
+                ind -= modulo[0]
+            else:
+                offsets = ind // modulo[1:] % sizes
```

Comment:
not sure if there is a helpful comment that can be done on this line. took me a minute but alright.